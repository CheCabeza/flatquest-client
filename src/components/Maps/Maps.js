import * as React from 'react';
import {PropTypes} from 'prop-types';
import {connect} from 'react-redux';
import {StyleSheet} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import MapView, {PROVIDER_GOOGLE, Marker, Callout} from 'react-native-maps';
import FlatCardById from '../FlatCardById/FlatCardById';

function Maps({flats}) {
  const navigation = useNavigation();
  let _mapView;

  return (
    <MapView
      style={mapStyles.mapContainer}
      provider={PROVIDER_GOOGLE}
      ref={mapView => {
        _mapView = mapView;
      }}
      initialCamera={{
        center: {latitude: 41.396519, longitude: 2.197323},
        pitch: 0,
        zoom: 14,
        heading: 0,
        altitude: 0,
      }}>
      {flats.map((flatData, index) => (
        <Marker
          key={index}
          coordinate={{
            latitude: +flatData.coordinates.latitude,
            longitude: +flatData.coordinates.longitude,
          }}
          onPress={() =>
            _mapView.animateCamera({
              center: {
                latitude: +flatData.coordinates.latitude,
                longitude: +flatData.coordinates.longitude,
              },
              pitch: 0,
              zoom: 14,
              heading: 0,
              altitude: 0,
            })
          }
          title={flatData.address}>
          <Callout
            testID="mapButton"
            onPress={() =>
              navigation.navigate('Details', {flatId: flatData._id})
            }>
            <FlatCardById flatId={flatData._id} />
          </Callout>
        </Marker>
      ))}
    </MapView>
  );
}
Maps.propTypes = {
  flats: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(store) {
  return {
    flats: store.flats,
    user: store.user,
  };
}

export default connect(mapStateToProps)(Maps);

const mapStyles = StyleSheet.create({
  mapContainer: {
    height: '100%',
  },
});
