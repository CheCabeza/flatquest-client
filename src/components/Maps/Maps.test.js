import React from 'react';
import {render} from '@testing-library/react-native';
import {useNavigation} from '@react-navigation/native';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Maps from './Maps';
import thunk from 'redux-thunk';

jest.mock('@react-navigation/native');

describe('Given a Maps component', () => {
  const mockStore = configureStore([thunk]);
  let store;

  beforeEach(() => {
    store = mockStore({
      user: {
        user: {
          email: 'ho@la.com',
          pass: '123123123',
          favorites: [123],
        },
      },
      flats: [
        {
          _id: 123,
          photos: 'a',
          pros: ['glass', 'folder-open'],
          cons: ['money', 'copy'],
          reviews: ['Mierda de test'],
          coordinates: {latitude: 1, longitude: 2},
        },
      ],
    });
    useNavigation.mockReturnValueOnce({
      navigate: jest.fn(),
    });
  });

  test('Should render the Maps of page', () => {
    const lobby = render(
      <Provider store={store}>
        <Maps />
      </Provider>,
    );
    expect(lobby).toMatchSnapshot();
  });
});
