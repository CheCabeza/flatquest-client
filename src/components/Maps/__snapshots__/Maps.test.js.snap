// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Maps component Should render the Maps of page 1`] = `null`;
